functions/visualizations.py

```diff
@@ -35,15 +36,42 @@ def vis_preprocessing(event: dict, context: BackgroundContext):
         session.commit()
 
 
-def _get_file_from_gcs(bucket_name: str, object_name: str) -> BytesIO:
+def _get_data_file(
+    object_name: str, as_string: bool = False
+) -> Union[BytesIO, StringIO]:
     """Download data from GCS to a byte stream and return it."""
     storage_client = storage.Client()
-    bucket = storage_client.get_bucket(bucket_name)
+    bucket = storage_client.get_bucket(GOOGLE_DATA_BUCKET)
     blob = bucket.get_blob(object_name)
     file_str = blob.download_as_string()
+    if as_string:
+        return StringIO(file_str)
     return BytesIO(bytes(file_str))
 
 
+def _get_metadata_df(trial_id: str) -> pd.DataFrame:
+    """
+    Build a dataframe containing the participant/sample metadata for this trial,
+    joined on CIMAC ID and indexed on CIMAC ID.
+    """
+    participants_blob = _get_data_file(f"{trial_id}/participants.csv", as_string=True)
+    samples_blob = _get_data_file(f"{trial_id}/samples.csv", as_string=True)
+
+    participants_df = pd.read_csv(participants_blob)
+    samples_df = pd.read_csv(samples_blob)
+
+    metadata_df = pd.merge(
```

Comment:
I'm not sure, but my understanding was that samples file includes everything from the participants csv. It isn't?

Reply:
The participants CSV contains a 'cohort_name' column, which the samples CSV does not have in it. Maybe we should just be creating a single flattened samples CSV for each trial though.